From 82f0818a896b90c3bc683332bafb63eb6bd589df Mon Sep 17 00:00:00 2001
From: Tomaz Zaman <tomaz@mono.si>
Date: Sun, 9 Aug 2026 13:56:19 +0200
Subject: [PATCH] ipv6: addrconf: give a modified prefix route an absolute
 expiry

Backport upstream a7bfaba4823e3c165bb2004c74eff7c096672bc7 by Qishuai Liu:
convert a modified IPv6 prefix route's relative clock_t lifetime into an
absolute jiffies deadline, as the route creation path already does.

Signed-off-by: Tomaz Zaman <tomaz@mono.si>
---
 net/ipv6/addrconf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/ipv6/addrconf.c b/net/ipv6/addrconf.c
index fb239e8f8359..33439672b071 100644
--- a/net/ipv6/addrconf.c
+++ b/net/ipv6/addrconf.c
@@ -4873,7 +4873,7 @@ static int modify_prefix_route(struct net *net, struct inet6_ifaddr *ifp,
 			fib6_clean_expires(f6i);
 			fib6_may_remove_gc_list(net, f6i);
 		} else {
-			fib6_set_expires(f6i, expires);
+			fib6_set_expires(f6i, jiffies + clock_t_to_jiffies(expires));
 			fib6_add_gc_list(f6i);
 		}
 
-- 
2.47.3

